logos/benchmarks/benchmark_logos.py

```
@@ -2082,9 +2087,119 @@ def _overall_energy_metrics(tracker, t_start: float, t_end: float, n_ok: int, n_
+    Replaces the naive overlapping-window values computed inline during dispatch
+    (kept only for the live ``E=…J`` readout) so the CSV, charts, and per-request
+    stats reflect a fair split of the measured power. Applied per source (gpu /
+    wall) using that source's aggregate (node-summed) power timeline."""
+    sources = getattr(tracker, "children", None) or {"gpu": tracker}
+    for name, attr in (("gpu", "energy_gpu_j"), ("wall", "energy_wall_j")):
+        child = sources.get(name)
+        if child is None or not getattr(child, "available", False):
+            continue
+        try:
+            samples = child.power_samples()
+        except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)